bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -459,25 +459,39 @@ object BagsManager {
       }
     }
 
-    obs.getProgram.removeStructureChangeListener(StructureListener)
-    obs.getProgram.removeCompositeChangeListener(ChangeListener)
-    applySelection(TpeContext(obs))
-
-    // Update the TPE if it is visible
-    selOpt.foreach { sel =>
-      Option(TpeManager.get()).filter(_.isVisible).foreach { tpe =>
-        sel.assignments.foreach { ass =>
-          val clazz = ass.guideProbe.getType match {
-            case GuideProbe.Type.AOWFS => classOf[Altair_WFS_Feature]
-            case GuideProbe.Type.OIWFS => classOf[OIWFS_Feature]
-            case GuideProbe.Type.PWFS  => classOf[TpePWFSFeature]
+    val ctx = TpeContext(obs)
+
+    // If the PA is not allowed to change by the PA constraint, but does, then this is an intermediate
+    // result that must be ignored: further changes to the PA have been made since this result was obtained,
+    // so do not process this case.
+    val process = (for {
+      paCst <- ctx.obsContext.map(_.getPosAngleConstraint)
+      oldPA <- ctx.instrument.dataObject.map(_.getPosAngleDegrees)
+      newPA <- selOpt.map(_.posAngle.toDegrees)
+    } yield paCst != PosAngleConstraint.FIXED || oldPA == newPA) | false
```

Comment:
So here, the `process` flag could be set based upon whether the current hash matches the expected one which would be passed into `applyAction`.  It's the same idea but it applies to any changes that impact the AGS calculation rather than just the pos angle constraint.